packages/cli/src/commands/source/push/delete.ts

```diff
@@ -0,0 +1,165 @@
+import {Command, flags} from '@oclif/command';
+import {Source} from '@coveo/push-api-client';
+import {AuthenticatedClient} from '../../../lib/platform/authenticatedClient';
+import {
+  IsAuthenticated,
+  Preconditions,
+} from '../../../lib/decorators/preconditions';
+import dedent from 'ts-dedent';
+import {
+  buildAnalyticsFailureHook,
+  buildAnalyticsSuccessHook,
+} from '../../../hooks/analytics/analytics';
+import {green, red} from 'chalk';
+// eslint-disable-next-line node/no-extraneous-import
+import {AxiosResponse} from 'axios';
+
+interface ErrorFromAPI {
+  response: {
+    status: number;
+    data: {
+      errorCode: string;
+      message: string;
+    };
+  };
+}
```

Comment:
That seems pretty generic for an error, do you think we should extract it in an utils or will it be a one-use interface? 🤔

Reply:
Good point.

I think this will also be used for `:push:add`.

However, I'll do the change/extract shared interface for both when I work on that command.